packages/host/app/services/realm.ts

```diff
@@ -22,21 +25,25 @@ import {
   JWTPayload,
   SupportedMimeType,
   type RealmInfo,
+  type IndexEventData,
 } from '@cardstack/runtime-common';
 
 import ENV from '@cardstack/host/config/environment';
 
+import { assertNever } from '@cardstack/host/utils/assert-never';
+
 import type MatrixService from './matrix-service';
+import type MessageService from './message-service';
 import type NetworkService from './network';
 import type RealmServerService from './realm-server';
 import type ResetService from './reset';
 
 const log = logger('service:realm');
 
-interface Meta {
-  info: RealmInfo;
+export type EnhancedRealmInfo = RealmInfo & {
+  isIndexing: boolean;
   isPublic: boolean;
-}
+};
```

Comment:
In order to prevent having nested tracked objects, I flattened the `Meta` interface (which only had a single extra `isPublic` field into the `RealmInfo` interface, renaming it `ExtendedRealmInfo`.